src/app/components/plugin-uiframe/plugin-uiframe.component.ts

```diff
@@ -307,23 +310,28 @@ export class PluginUiframeComponent implements OnChanges, OnDestroy {
             mergeAll(),
             map(wholePage => 
                 wholePage.pipe(
-                    map(apiObjectList => apiObjectList.items.filter(experminetData => experminetData.contentType === "application/qasm" || experminetData.contentType === "application/qiskit")),
+                    map(apiObjectList => apiObjectList.items.filter(experminetData => allowedContentTypes.has(experminetData.contentType))),
                     map(dataItems => dataItems.map(item => this.experimentId ? this.backend.getExperimentData(this.experimentId, item.name, item.version) : undefined)),
-                    filter((experimentData): experimentData is Observable<ExperimentDataApiObject>[] => !!experimentData),
+                    filter((experimentData): experimentData is Observable<ExperimentDataApiObject>[] => Boolean(experimentData)),
                     mergeAll(),
                     concatAll(),
                 )
             ),
             concatAll(),
-            map(dataItem => (this.experimentId && dataItem.producedBy) ?
-                this.backend.getTimelineStep(this.experimentId, dataItem.producedBy).pipe(
-                    map(step => ({
-                        name: dataItem.name + ' ' + step.processorName,
-                        download: dataItem.download,
-                        version: dataItem.version,
-                        type: dataItem.contentType.split("/")[1]
-                    })),
-                ) : of(undefined)),
+            map(dataItem => {
+                if (this.experimentId && dataItem.producedBy) {
+                    return this.backend.getTimelineStep(this.experimentId, dataItem.producedBy).pipe(
+                        map(step => ({
+                            name: dataItem.name + ' ' + step.processorName,
+                            download: dataItem.download,
+                            version: dataItem.version,
+                            type: contentTypeToType(dataItem.contentType)
```

Comment:
Include logic to handle the case where the content type is not in the map (if such logic is required)